File: KITE-Simulcast-Test/src/main/java/io/cosmosoftware/kite/simulcast/steps/SelectProfileStep.java
package io.cosmosoftware.kite.simulcast.steps;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.simulcast.pages.SimulcastPageBase;
import io.cosmosoftware.kite.steps.TestStep;
import org.openqa.selenium.WebDriver;

import static io.cosmosoftware.kite.entities.Timeouts.ONE_SECOND_INTERVAL;
import static io.cosmosoftware.kite.util.TestUtils.waitAround;

public class SelectProfileStep extends TestStep {

  private final SimulcastPageBase loopbackPage;

  private final String rid;
  private final int tid;

  public SelectProfileStep(Runner runner, SimulcastPageBase page, String rid, int tid) {
    super(runner);
    this.loopbackPage = page;
    this.rid = rid;
    this.tid = tid;
  }
  
  @Override
  public String stepDescription() {
    return "Clicking button " + rid + tid;
  }
  
  @Override
  protected void step() throws KiteTestException {
    loopbackPage.clickButton(rid, tid);
    waitAround(3 * ONE_SECOND_INTERVAL);
  }
}

File: KITE-Simulcast-Test/js/steps/SelectProfileStep.js
const {TestStep} = require('kite-common');

/**
 * Class: SelectProfileStep
 * Extends: TestStep
 * Description:
 */
class SelectProfileStep extends TestStep {
  constructor(kiteBaseTest, rid, tid) {
    super();
    this.driver = kiteBaseTest.driver;
    this.rid = rid;
    this.tid = tid;
    this.statsCollectionInterval = kiteBaseTest.statsCollectionInterval;
    this.page = kiteBaseTest.page;
  }

  stepDescription() {
    return 'Choose a profile id gauge : ' + this.rid + ' ' + this.tid;
  }

  async step() {
    await this.page.selectProfile(this);
  }
}

module.exports = SelectProfileStep;